pkg/cli/image/manifest/dockercredentials/auth_resolver_test.go

```diff
@@ -0,0 +1,88 @@
+package dockercredentials
+
+import (
+	"github.com/containers/image/v5/docker/reference"
+	"testing"
+
+	"github.com/openshift/oc/pkg/helpers/image/credentialprovider"
+)
+
+func Test_AuthResolver(t *testing.T) {
+	fn := func(host string, entry credentialprovider.DockerConfigEntry) AuthResolver {
+		return AuthResolver{
+			map[string]credentialprovider.DockerConfigEntry{
+				host: entry,
+			},
+		}
+	}
+	def := credentialprovider.DockerConfigEntry{
+		Username: "local_user",
+		Password: "local_pass",
+	}
+
+	type testInput struct {
+		authResolver AuthResolver
+		image        string
+	}
+	tests := []struct {
+		name     string
+		input    testInput
+		user     string
+		password string
+	}{
+		{name: "docker/docker,", input: testInput{authResolver: fn("index.docker.io", def), image: "docker/docker"}, user: def.Username, password: def.Password},
+		{name: "library/debian", input: testInput{authResolver: fn("docker.io", def), image: "library/debian"}, user: def.Username, password: def.Password},
```

Comment:
should it be `fn("index.docker.io", def)`?

Reply:
this one is testing the old docker format